### PR TITLE
Javadoc improvements

### PR DESCRIPTION
**Pull request by pshipton**
- replace IBM with OpenJ9
- fix some links
- add missing package-info

Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
Pushed an update.

**Comment by pshipton** [comment]
Pushed another tweak.

File: jcl/src/jdk.management/share/classes/com/sun/management/HotSpotDiagnosticMXBean.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 import java.lang.management.PlatformManagedObject;
 
 /**
- * Stub class to compile OpenJDK diagnostic management, see openj9.lang.management and com.ibm.lang.management packages instead.
+ * Stub class to compile OpenJDK diagnostic management, see {@link openj9.lang.management} and {@link com.ibm.lang.management} packages instead.
```
Comment (keithc-ca) [inline jcl/src/jdk.management/share/classes/com/sun/management/HotSpotDiagnosticMXBean.java:29]: Perhaps change 'class' to 'interface'?

**Comment by pshipton** [comment]
Pushed new content to address all the review comments.

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/OperatingSystemMXBean.java [hunk 2]
```diff
@@ -24,7 +24,7 @@
 package com.ibm.lang.management;
 
 /**
- * IBM specific platform management interface for the Operating System on which the Java Virtual Machine is running. 
+ * OpenJ9 platform management interface for the Operating System on which the Java Virtual Machine is running.
```
[the remark this answers is not in the document]
Comment (keithc-ca) [reply jcl/src/jdk.management/share/classes/com/ibm/lang/management/OperatingSystemMXBean.java]: Better (it seems the leading whitespace is lost on the first line):
```
 ...
  com.ibm.lang.management.OperatingSystemMXBean osmxbean = null;
  osmxbean = (com.ibm.lang.management.OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
  ...
```
Apparently, it needs to be:
```
 * {@code
 * ...
 * com.ibm.lang.management.OperatingSystemMXBean osmxbean = null;
 * osmxbean = (com.ibm.lang.management.OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
 * ...
 * }
```

[the remark this answers is not in the document]
Comment (pshipton) [reply jcl/src/jdk.management/share/classes/com/ibm/lang/management/OperatingSystemMXBean.java]: updated

**Pull request merged**